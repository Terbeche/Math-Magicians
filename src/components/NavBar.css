.Nav{
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 1rem 3rem;
    margin: 2rem;
background-color: antiquewhite;
}


.Nav-links{
    display: flex;
    justify-content: space-around;
    gap: 5rem;
}

ul{
    list-style-type: none;
}

a {
    text-decoration: none;
}

.Nav-brand{
    font-size: 3rem;
    color: blueviolet;
}

.Nav-item{
    font-size: 1.5rem;
}

.Nav-link{
    color: blue;
}