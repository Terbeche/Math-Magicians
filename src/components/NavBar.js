/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import './NavBar.css';
import { Link } from 'react-router-dom';
import { MdClose } from 'react-icons/md';
import { FiMenu } from 'react-icons/fi';

const NavBar = () => {
  const [navbarOpen, setNavbarOpen] = useState(false);

  const handleToggle = () => {
    setNavbarOpen((prev) => !prev);
  };

  const closeMenu = () => {
    setNavbarOpen(false);
  };
  return (
    <nav className="Nav">
      <button type="button" onClick={handleToggle}>
        {navbarOpen ? (
          <MdClose style={{ color: 'red', width: '40px', height: '40px' }} />
        ) : (
          <FiMenu style={{ color: '#7b7b7b', width: '40px', height: '40px' }} />
        )}
      </button>
      <Link
        onClick={() => closeMenu()}
        to="/"
        className="Nav-brand"
      >
        Math Magicians
      </Link>
      <ul className={`Nav-links menuNav ${navbarOpen ? ' showMenu' : ''}`}>
        <li className="Nav-item">
          <Link
            onClick={() => closeMenu()}
            className="Nav-link"
            to="/"
          >
            Home
          </Link>
        </li>
        <li className="Nav-item">
          <Link
            onClick={() => closeMenu()}
            className="Nav-link"
            to="/calculator"
          >
            Calculator
          </Link>
        </li>
        <li className="Nav-item">
          <Link
            onClick={() => closeMenu()}
            className="Nav-link"
            to="/quote"
          >
            Quote
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default NavBar;
