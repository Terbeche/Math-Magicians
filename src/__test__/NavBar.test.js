import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('Testing the NavBar', () => {
  it('NavBar is rendered', () => {
    const nav = renderer
      .create(<BrowserRouter><NavBar /></BrowserRouter>)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
