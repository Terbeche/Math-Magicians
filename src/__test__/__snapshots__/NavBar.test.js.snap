// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the NavBar NavBar is rendered 1`] = `
<nav
  className="Nav"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "#7b7b7b",
          "height": "40px",
          "width": "40px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </button>
  <a
    className="Nav-brand"
    href="/"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul
    className="Nav-links menuNav "
  >
    <li
      className="Nav-item"
    >
      <a
        className="Nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="Nav-item"
    >
      <a
        className="Nav-link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="Nav-item"
    >
      <a
        className="Nav-link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
